lib/phoenix/transports/json_serializer.ex

```diff
@@ -4,17 +4,38 @@ defmodule Phoenix.Transports.JSONSerializer do
 
   @behaviour Phoenix.Transports.Serializer
 
+  alias Phoenix.Socket.Reply
+  alias Phoenix.Socket.Message
+  alias Phoenix.Socket.Broadcast
+
   @doc """
   Encodes a `Phoenix.Socket.Message` struct to JSON string.
   """
-  def encode!(message), do: {:text, Poison.encode_to_iodata!(message)}
```

Comment:
I would rename this module to websocket_serializer. It just happens to use JSON (which we would document.